apps/api-extractor/src/api/model/ApiItem.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { Constructor, PropertiesOf } from '../mixins/Mixin';
+
+/** @public */
+export const enum ApiItemKind {
+  Class = 'Class',
+  EntryPoint = 'EntryPoint',
+  Interface = 'Interface',
+  Method = 'Method',
+  MethodSignature = 'MethodSignature',
+  Model = 'Model',
+  Namespace = 'Namespace',
+  Package = 'Package',
+  Parameter = 'Parameter',
+  Property = 'Property',
+  PropertySignature = 'PropertySignature',
+  None = 'None'
+}
+
+/** @public */
+export interface IApiItemOptions {
+  name: string;
+}
+
+export interface IApiItemJson {
+  name: string;
+  kind: ApiItemKind;
+  canonicalReference: string;
+}
+
+/**
+ * PRIVATE
+ * Allows ApiItemContainerMixin to assign the parent.
+ */
+// tslint:disable-next-line:variable-name
+export const ApiItem_parent: unique symbol = Symbol('ApiItem._parent');
+
+/** @public */
+export class ApiItem {
```

Comment:
>ApiItem [](start = 13, length = 7)

It might make sense to make `IApiItemOptions` and `IApiItemJson` generic types on this class so you don't have to re-specify `serializeInto` in every child interface.

Reply:
@iclanton How would that work with mixins?  I don't think they can supply a generic parameter easily.

---
In reply to: [236465944](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236465944) [](ancestors = 236465944)